### PR TITLE
Adding lib unit tests

### PR DESCRIPTION
**Pull request by WhatIsHeDoing**
I love this plugin, and noticed PRs were welcome, so thought I'd start learning the codebase by adding unit tests. Hope this actually helps!

Started with lib utils and believe the tests identify and fix some small edge cases. Happy to expand the coverage if this is the right direction.

### CODE REVIEW

File: lib/README.md [hunk 1]
```diff
@@ -6,11 +6,30 @@ Text extraction does not work on mobile; calling the functions will just immedia
 
 ## Build
 
-You'll need [Rust](https://doc.rust-lang.org/book/ch01-01-installation.html), [wasm-pack](https://github.com/rustwasm/wasm-pack), and [pnpm](https://pnpm.io/installation). 
+You'll need [Rust], [wasm-pack], and [pnpm].
 
 ```sh
-$ pnpm i
-$ pnpm run build
+pnpm i
```
Comment (scambier) [inline lib/README.md]: The `$` is a convention to indicate the command is meant to be run as a normal user (as opposed to `#` for a root user). It's also a useful visual indicator when long lines wrap around, though it's not the case here :)

File: lib/README.md [hunk 2]
```diff
@@ -6,11 +6,30 @@ Text extraction does not work on mobile; calling the functions will just immedia
...
-Rust is quite slow to compile, so the first build will take some time.
\ No newline at end of file
+Rust is quite slow to compile, so the first build will take some time.
+
+## Test
+
+Unit tests use the fast [Vitest] library, and sit alongside code in the format `<name>.test.ts`. Run them continuously with:
+
+```sh
+pnpm test
```
Comment (scambier) [inline lib/README.md]: $ here too

File: lib/README.md [hunk 3]
```diff
@@ -6,11 +6,30 @@ Text extraction does not work on mobile; calling the functions will just immedia
...
+```
+
+Test coverage can be shown with:
+
+```sh
+pnpm coverage
```
Comment (scambier) [inline lib/README.md]: $ here too

**Review by scambier — CHANGES_REQUESTED** [review]
Thanks for your PR 👍

Could you just re-add the `$` in front of the command lines? Thank you.

**Comment by WhatIsHeDoing** [comment]
Ah no problem, I removed them as they were highlighted by the markdownlint [MD014](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md014.md) warning.

**Comment by scambier** [comment]
Thanks 👍
That's quite an opinionated linter 😅
